firebase-database/src/main/java/com/google/firebase/database/core/utilities/Utilities.java

```diff
@@ -70,36 +44,68 @@ public static ParsedUrl parseUrl(String url) throws DatabaseException {
       if (port != -1) {
         repoInfo.secure = scheme.equals("https");
         repoInfo.host += ":" + port;
+      } else if (repoInfo.host.equals("localhost")) {
```

Comment:
Why this change? Also what if people use `127.0.0.1` (a perfectly valid thing to do) instead of `localhost`?

Reply:
We can special-case `127.0.0.1` as well if you feel we should. This code just makes sure that for localhost we don't use SSL even if no port is specified. Without adding support for `127.0.0.1`, users need to specify `127.0.0.1:9000` (so really this only becomes an issue if the emulator runs on port 80 or 443)

Reply:
I think the existing behavior when port is present is fine. For other cases, it should be tied to the same condition as below -- if we decide to use namespace from host, then we force SSL (same behavior as before), or we use `scheme.equals("https")` if we decide to use ns from query. In this way all emulator-specific behavior is at least triggered by the same conditions and we don't have to think about such edge cases.

Reply:
As stated in the iOS port, this special case is now a thing of the past.